frontend/packages/console-shared/src/components/catalog/details/CatalogDetailsModal.tsx

```diff
@@ -33,43 +40,111 @@ const CatalogDetailsModal: React.FC<CatalogDetailsModalProps> = ({ item, onClose
 
   const vendor = <div>{provider}</div>;
 
+  const getButtonVariant = (variant?: string): ButtonVariant => {
+    switch (variant) {
+      case 'secondary':
+        return ButtonVariant.secondary;
+      case 'link':
+        return ButtonVariant.link;
+      default:
+        return ButtonVariant.primary;
+    }
+  };
+
+  const renderAction = (ctaLink, index: number) => {
+    const { to, label, callback, variant } = ctaLink;
+
+    if (!to && !callback) {
+      return null;
+    }
+
+    const handleClick = () => {
+      if (callback) {
+        callback();
+      }
+      onClose();
+    };
+
+    // If there's a callback and no href, render as a button
+    if (callback && !to) {
+      return (
+        <Button
+          key={index}
```

Comment:
the recommendation is to not use array indices for `key`s when possible for performance 

```suggestion
          key={`${name}-cta-${index}`}
```